server/src/main/java/org/elasticsearch/common/xcontent/ChunkedToXContentBuilder.java

```diff
@@ -246,16 +306,51 @@ public ChunkedToXContentBuilder field(String name) {
         return this;
     }
 
+    public ChunkedToXContentBuilder field(String name, boolean value) {
+        addChunk((b, p) -> b.field(name, value));
+        return this;
+    }
+
+    public ChunkedToXContentBuilder field(String name, Boolean value) {
+        addChunk((b, p) -> b.field(name, value));
+        return this;
+    }
+
+    public ChunkedToXContentBuilder field(String name, int value) {
+        addChunk((b, p) -> b.field(name, value));
+        return this;
+    }
+
+    public ChunkedToXContentBuilder field(String name, Integer value) {
+        addChunk((b, p) -> b.field(name, value));
+        return this;
+    }
+
     public ChunkedToXContentBuilder field(String name, long value) {
         addChunk((b, p) -> b.field(name, value));
         return this;
     }
 
+    public ChunkedToXContentBuilder field(String name, Long value) {
+        addChunk((b, p) -> b.field(name, value));
+        return this;
+    }
+
     public ChunkedToXContentBuilder field(String name, String value) {
         addChunk((b, p) -> b.field(name, value));
         return this;
     }
 
+    public ChunkedToXContentBuilder field(String name, Enum<?> value) {
+        addChunk((b, p) -> b.field(name, value));
+        return this;
+    }
+
+    public ChunkedToXContentBuilder field(String name, ToXContent value) {
+        addChunk((b, p) -> b.field(name, value, params));
```

Comment:
Nit: no need to capture `this` (for `params`) here

```suggestion
        addChunk((b, p) -> b.field(name, value, p));
```